osquery/tables/system/darwin/asl_utils.cpp

```diff
@@ -0,0 +1,108 @@
+#include <boost/algorithm/string/replace.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include "asl_utils.h"
+
+namespace ba = boost::algorithm;
+
+namespace osquery {
+namespace tables {
+
+bool isNumeric(ColumnType coltype) {
+  switch (coltype) {
+  case INTEGER_TYPE:
+  case BIGINT_TYPE:
+  case UNSIGNED_BIGINT_TYPE:
+  case DOUBLE_TYPE:
+    return true;
+  default:
+    return false;
+  }
+}
+
+void addQueryOp(asl_object_t& query,
+                const std::string& key,
+                const std::string& value,
+                ConstraintOperator op,
+                ColumnType col_type) {
+  if (key == kExtraColumnKey) {
+    // ASL doesn't know about the 'Extra' column, so we can't do the matching
+    // through the ASL query. Do nothing here, and later SQLite's engine will
+    // do the match.
+    return;
+  }
+
+  // Only some queries can be supported through ASL, those that are not will
+  // just be performed later by the SQLite engine.
+  auto it = kSupportedAslOps.find(op);
+  if (it != kSupportedAslOps.end()) {
+    std::string modified_val = value;
+    uint32_t asl_op = it->second;
```

Comment:
I take it this is the preferred `std::map` access pattern, so I'll change and prefer this in the future.